src/components/LayoutTree.tsx

```diff
@@ -470,17 +491,29 @@ const LayoutTree = () => {
   const {
     indentCursorAncestorTables,
     treeThoughtsPositioned,
+    hoverArrowVisibility,
   }: {
     // the global indent based on the depth of the cursor and how many ancestors are tables
     indentCursorAncestorTables: number
     treeThoughtsPositioned: TreeThoughtPositioned[]
+    hoverArrowVisibility: 'above' | 'below' | null
   } = useMemo(() => {
     // y increases monotically, so it is more efficent to accumulate than to calculate each time
     // x varies, so we calculate it each time
     // (it is especially hard to determine how much x is decreased on cliffs when there are any number of tables in between)
     let yaccum = 0
     let indentCursorAncestorTables = 0
 
+    // Arrow visibility based on insertionY
```

Comment:
This comment can be updated now that we're using the ranks rather than the `insertionY` calculation.